フロントエンド総合課題①

ryorta opened the pull request:
- ユーザー登録機能を実装する。
  - トークンを取得する。
- ログイン機能を実装する。
- ログアウト機能を実装する。
- ユーザー関連機能を実装する。
    - ユーザー情報を取得する。
    - ユーザー一覧を取得する。
    - ログインユーザーを削除する。
    - ユーザー情報を編集する。

ryorta committed:
- ユーザー登録機能を実装する。
    - トークンを取得する。
- ログイン機能を実装する。
- ログアウト機能を実装する。
- ユーザー関連機能を実装する。
    - ユーザー情報を取得する。
    - ユーザー一覧を取得する。
    - ログインユーザーを削除する。
    - ユーザー情報を編集する。

ghost commented:
- [x] /login
概要: トークンを発行する。自分の過去のトークンは全て失効する。
- [x] /register
概要: ユーザを新規登録する。トークンは発行されない。
name
- [x] 最大 16 文字
bio
- [x] 最大 128 文字
password
- [x] /logout
概要: 自分のトークンを全て失効させる。
- [x] /users/{id}
概要: ユーザを取得する。
- [x] /users
概要: ユーザ一覧を取得する。
- [x] q (任意)
name と bio をそれぞれあいまい検索するキーワード
- [x] /users
概要: ログインユーザを削除する。
- [x] /users
概要: ログインユーザを編集する。
- [x] bio
最大 128 文字

ryorta committed:
resisterのnameとbioの文字数制限の追加
usersEditのbioの文字数制限の追加
usersIdGetの想定外のID、空白の場合の判定
ログアウト時にローカルストレージからトークンの削除
URLSearchParams()を用いて、ユーザーリストの検索機能修正
テキストファイル最終行の改行

ghost commented:
- [x] /login

概要: トークンを発行する。自分の過去のトークンは全て失効する。
- [ ] /register
概要: ユーザを新規登録する。トークンは発行されない。
- [x] name
 最大 16 文字
- [x] bio
 最大 128 文字
password
- [ ]  /logout
概要: 自分のトークンを全て失効させる。
- [x]  /users/{id}
概要: ユーザを取得する。
- [x] /users
概要: ユーザ一覧を取得する。
- [x] q (任意)
name と bio をそれぞれあいまい検索するキーワード
- [x] /users
概要: ログインユーザを削除する。
- [ ] /users
概要: ログインユーザを編集する。
- [x] bio
最大 128 文字

ryorta committed:
catchによるエラー回避
logout処理による、トークンの削除
